src/build_system.ml

```diff
@@ -518,6 +518,8 @@ module Trace = struct
     trace
 end
 
+let load_trace () = Option.some_if (Sys.file_exists Trace.file) (Trace.load ())
```

Comment:
This doesn't really work. Both arguments will be evaluated before `Option.some_if` is called.